File: week17/week17-2.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int N= digits.size();
        int carry=1;
        for(int i=N-1;i>=0;i--){
            digits[i]=digits[i]+carry;
            if(digits[i]>=10){
                carry=1;
                digits[i]=digits[i]%10;
            }
            else{
                carry=0;
            }
        }
        if(carry==0) return digits;
        else{
            vector<int> ans(N+1);
            ans[0]=carry;
            for(int i=0;i<N;i++){
                ans[i+1]= digits[i];
            }
            return ans;
        }
    }
};

File: week17/week17-2.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        N=len(digits)
        carry=1
        for i in range(N-1,-1,-1):
            digits[i]=digits[i]+carry
            if digits[i]>=10:
                carry=1
                digits[i]=digits[i]%10
            else:
                carry=0
        if carry==0:
            return digits
        else:
            return [carry]+digits